### PR TITLE
Replace tsc with esbuild to build packages

### PR DESCRIPTION
**Pull request by TrySound**
Build of whole monorepo without cache takes 1min or more on my laptop. tsc is very slow tool for this. Also having rootDir in tsconfig prevents checking outside of src folder.

Here I added "scripts" package with "build-package" bin. It does both esm and cjs builds. --watch flag support completely replaces tsup for us.

Now full rebuild of monorepo without cache takes only 10s including prisma-client, http-client and designer builds.

## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [x] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [ ] conceptual
  - [x] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: packages/scripts/build-package.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env -S node --experimental-specifier-resolution=node --loader=tsx
```
Comment (kof) [inline packages/scripts/build-package.ts]: we have been running bin scripts always with tsx so far using tsx from scripts, I would love a consistency

File: packages/scripts/build-package.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+const entryPoints: string[] = [];
+const assets: string[] = [];
+
+await totalist("./src", (rel) => {
+  if (rel.endsWith(".ts") || rel.endsWith(".tsx")) {
+    if (rel.includes(".test.") || rel.includes(".stories.")) {
+      return;
+    }
+    entryPoints.push(join("src", rel));
```
Comment (Andarist) [inline packages/scripts/build-package.ts:19]: out of curiosity - why not flat-bundling? to cover for `/server`?

Comment (TrySound) [reply packages/scripts/build-package.ts:19]: Bundling breaks code splitting. All bundlers do it based on modules.

File: packages/scripts/build-package.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env tsx
```
Comment (kof) [inline packages/scripts/build-package.ts:1]: lets do it this way everywhere then?

Comment (kof) [reply packages/scripts/build-package.ts:1]: all bin/*

**Review by Andarist — APPROVED** [review]

File: apps/designer/package.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
     "build": "remix build",
     "dev": "remix dev",
     "start": "remix-serve build",
-    "typecheck": "tsc -b",
+    "typecheck": "tsc",
```
Comment (Andarist) [inline apps/designer/package.json:9]: nice, I've noticed that this was a leftover here but had no idea that it could impact the watch behaviour like this

**Review by kof — APPROVED** [review]

**Review by istarkov — APPROVED** [review]

**Pull request merged**